view/ViewableData.php

```diff
@@ -541,6 +542,9 @@ public function CSSClasses($stopAtClass = 'ViewableData') {
 			$classes[] = $this->template;
 		}
 
+		// Strip out namespaces
+		$classes = preg_replace('#.*\\\\#', '', $classes);
```

Comment:
I think the CSS classes for "MySite\Page" should probably be "Page" and "Mysite-Page"? Otherwise we're not solving the name clash problem for CSS.

(Also, preg_replace is ick. Although it's more work, pushing this into ClassInfo::ancestry is nicer. Might actually be faster too - http://stackoverflow.com/questions/19901850/how-do-i-get-an-objects-unqualified-short-class-name#answer-25472778)